### PR TITLE
feat: 🍰 Implement Group GQL Model And CRUD Resolvers – First Step

### LINKED ISSUE #5131: 🚀 [Feature] Implement Group GraphQL Model And CRUD Resolvers In Backend
## :rocket: Feature Request

Implement group GraphQL model and CRUD resolvers.

- relates #5059

### Todo

- [x] GraphQL
  - [x] add group settings ( details see https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/5059#user-content-group-settings )
    - [x] owner/admin
    - [x] name (title)
    - [x] slug-name
    - [x] about (goal)
    - [x] description:
      - [x] minimal 100 chars
    - [x] group type
    - [x] action radius
    - [x] location
    - [x] categories: 1-3
    - [x] add relation `CREATED`
  - [x] rename `UserGroup` to `UserRole`
  - [x] define resolver for:
    - [x] `CreateGroup`
      - [x] error on name length, see frontend
    - [x] `Group`
    - [x] `UpdateGroup`
    - [x] `DeleteGroup`, see #5388
- [x] seeding:
  - [x] seed a group

### PR DESCRIPTION
**Pull request by Tirokk**
## 🍰 Pullrequest

Implement group GQL model and CRUD resolvers – first step.

### Issues

- relates #5131
- relates #5059

### Todo

- [x] @Mogge suggestions:
  - [x] `international` → `global`
  - [x] remove `wasSeede`
- [ ] GraphQL
  - [ ] add group settings ( details see https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/5059#user-content-group-settings )
    - [x] owner/admin
    - [x] name (title)
    - [x] slug-name
    - [x] about (goal)
    - [x] description:
      - [x] minimal 100 chars
    - [x] group type
    - [x] action radius
    - [ ] location
    - [x] categories: 1-3
    - [x] add relation `CREATED`
  - [x] rename `UserGroup` to `UserRole`
  - [ ] define resolver for:
    - [ ] `AvailableGroupTypes`
    - [ ] `AvailableGroupActionRadii`
    - [ ] `AvailableGroupMemberRoles`?
    - [ ] `UpdateGroup`
    - [ ] `DeleteGroup`
  - [ ] `Deletable`:
    - [ ] What happens with their groups if users deletes their user profile?
- [x] resolvers:
  - [x] CreateGroup:
    - [x] test correct slugification
  - [x] check for tests in:
    - [x] backend/src/middleware/permissionsMiddleware.spec.js
    - [x] backend/src/middleware/slugifyMiddleware.spec.js
- [ ] seeding:
  - [ ] seed a group
- [x] Neo4j database:
  - [x] how to add a `slug` contraint for `Group` label?
    - [x] where is the `slug` contraint for `Post` and `User` label? See `backend/src/db/migrate/store.js`
- [x] cleanup: search for `Wolle`

### CODE REVIEW

**Review by Elweyn — COMMENTED** [review]
Read it and the road seems good!

**Comment by Tirokk** [comment]
After @Mogge suggestion I do the remainders in the next PR 🍀🙏🏼

File: backend/src/schema/resolvers/groups.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+              OPTIONAL MATCH (:User {id: $userId})-[membership:MEMBER_OF]->(group)
+              RETURN group {.*, myRole: membership.role}
+            `
+          }
+        }
+        const result = await txc.run(groupCypher, {
+          userId: context.user.id,
+        })
+        const group = result.records.map((record) => record.get('group'))
+        return group
```
Comment (Mogge) [inline backend/src/schema/resolvers/groups.js:40]:
```suggestion
        return result.records.map((record) => record.get('group'))
```

File: backend/src/schema/resolvers/groups.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      })
+      try {
+        const group = await readTxResultPromise
+        return group
```
Comment (Mogge) [inline backend/src/schema/resolvers/groups.js:44]:
```suggestion
        return await readTxResultPromise
```

File: backend/src/schema/resolvers/groups.spec.js [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+          about: 'We will change the world!',
+          description: 'Some description' + descriptionAddition100,
+          groupType: 'public',
+          actionRadius: 'regional',
+          categoryIds,
+        },
+      })
+    })
+
+    describe('can find', () => {
```
Comment (Mogge) [inline backend/src/schema/resolvers/groups.spec.js:142]:
```suggestion
    describe('query groups', () => {
```

File: backend/src/schema/resolvers/groups.spec.js [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+      it('all', async () => {
```
Comment (Mogge) [inline backend/src/schema/resolvers/groups.spec.js:143]:
```suggestion
      describe('without any filters', () => {
        it('finds all groups', async () => {
```

File: backend/src/schema/resolvers/groups.spec.js [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+                myRole: null,
+              }),
+            ]),
+          },
+          errors: undefined,
+        }
+        await expect(query({ query: groupQuery, variables: {} })).resolves.toMatchObject(expected)
+      })
+
+      it('where user is member (or owner in this case)', async () => {
```
Comment (Mogge) [inline backend/src/schema/resolvers/groups.spec.js:164]:
```suggestion
      decribe(isMember = true', () => {
        it('finds only groups where user is member', async () => {
```

File: backend/src/schema/resolvers/groups.spec.js [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+            ],
+          },
+          errors: undefined,
+        }
+        await expect(
+          query({ query: groupQuery, variables: { isMember: true } }),
+        ).resolves.toMatchObject(expected)
+      })
+
+      it('where user is not(!) member', async () => {
```
Comment (Mogge) [inline backend/src/schema/resolvers/groups.spec.js:182]:
```suggestion
      decribe(isMember = false', () => {
        it('finds only groups where user is not(!) member', async () => {
```

File: backend/src/schema/resolvers/groups.spec.js [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+          },
+        },
+        errors: undefined,
+      }
+      await expect(mutate({ mutation: createGroupMutation, variables })).resolves.toMatchObject(
+        expected,
+      )
+    })
+
+    it('"disabled" and "deleted" default to "false"', async () => {
```
Comment (Mogge) [inline backend/src/schema/resolvers/groups.spec.js:261]:
```suggestion
    it('has "disabled" and "deleted" default to "false"', async () => {
```

File: backend/src/schema/resolvers/groups.js [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+        session.close()
+      }
+    },
+  },
+  Mutation: {
+    CreateGroup: async (_parent, params, context, _resolveInfo) => {
+      const { categoryIds } = params
+      delete params.categoryIds
+      if (!categoryIds || categoryIds.length < CATEGORIES_MIN) {
+        throw new UserInputError('To Less Categories!')
```
Comment (Mogge) [inline backend/src/schema/resolvers/groups.js:57]:
```suggestion
        throw new UserInputError('Too view categories!')
```

File: backend/src/schema/resolvers/groups.js [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+      }
+      if (categoryIds && categoryIds.length > CATEGORIES_MAX) {
+        throw new UserInputError('To Many Categories!')
+      }
+      if (
+        params.description === undefined ||
+        params.description === null ||
+        removeHtmlTags(params.description).length < DESCRIPTION_WITHOUT_HTML_LENGTH_MIN
+      ) {
+        throw new UserInputError('To Short Description!')
```
Comment (Mogge) [inline backend/src/schema/resolvers/groups.js:67]:
```suggestion
        throw new UserInputError('Description too short!')
```

File: backend/src/schema/resolvers/groups.js [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+  },
+  Mutation: {
+    CreateGroup: async (_parent, params, context, _resolveInfo) => {
+      const { categoryIds } = params
+      delete params.categoryIds
+      if (!categoryIds || categoryIds.length < CATEGORIES_MIN) {
+        throw new UserInputError('To Less Categories!')
+      }
+      if (categoryIds && categoryIds.length > CATEGORIES_MAX) {
+        throw new UserInputError('To Many Categories!')
```
Comment (Mogge) [inline backend/src/schema/resolvers/groups.js:60]:
```suggestion
        throw new UserInputError('Too many categories!')
```

File: backend/src/schema/types/enum/GroupActionRadius.gql [hunk 11]
```diff
@@ -0,0 +1,6 @@
+enum GroupActionRadius {
+  regional
+  national
+  continental
+  international
```
Comment (Mogge) [inline backend/src/schema/types/enum/GroupActionRadius.gql:5]:
```suggestion
  global
  interplanetary
```

**Review by Mogge — CHANGES_REQUESTED** [review]

File: backend/src/schema/types/enum/GroupActionRadius.gql (hunk 11 above)
Comment (Tirokk) [reply backend/src/schema/types/enum/GroupActionRadius.gql:5]: Done

File: backend/src/schema/resolvers/groups.js (hunk 10 above)
Comment (Tirokk) [reply backend/src/schema/resolvers/groups.js:58]: Done

File: backend/src/schema/resolvers/groups.js (hunk 9 above)
Comment (Tirokk) [reply backend/src/schema/resolvers/groups.js:65]: Done

File: backend/src/schema/resolvers/groups.js (hunk 8 above)
Comment (Tirokk) [reply backend/src/schema/resolvers/groups.js:55]: Done

File: backend/src/schema/resolvers/groups.spec.js (hunk 7 above)
Comment (Tirokk) [reply backend/src/schema/resolvers/groups.spec.js:261]: Done

File: backend/src/schema/resolvers/groups.spec.js (hunk 6 above)
Comment (Tirokk) [reply backend/src/schema/resolvers/groups.spec.js:182]: Done

File: backend/src/schema/resolvers/groups.spec.js (hunk 5 above)
Comment (Tirokk) [reply backend/src/schema/resolvers/groups.spec.js:164]: Done

File: backend/src/schema/resolvers/groups.spec.js (hunk 4 above)
Comment (Tirokk) [reply backend/src/schema/resolvers/groups.spec.js:143]: Done

File: backend/src/schema/resolvers/groups.spec.js (hunk 3 above)
Comment (Tirokk) [reply backend/src/schema/resolvers/groups.spec.js:142]: Done

**Review by Mogge — APPROVED** [review]

**Review by Mogge — APPROVED** [review]

**Pull request merged**